src/Groups/cosets.jl

```diff
@@ -575,6 +575,24 @@ function double_cosets(G::T, H::GAPGroup, K::GAPGroup; check::Bool=true) where T
    #return [GroupDoubleCoset(G,H,K,group_element(G.X,GAPWrap.Representative(dc)),dc) for dc in dcs]
 end
 
+raw"""
+    double_cosets_representatives_and_sizes(G::GAPGroup, H::GAPGroup, K::GAPGroup)
+    
+Return representatives of the double coset `H\G/K` and their size. 
+
+This function is faster and consumes less memory than [`double_cosets`](@ref)
+since no double cosets are created and stored.    
+"""
+function double_cosets_representatives_and_sizes(G::T, H::GAPGroup, K::GAPGroup) where T <: GAPGroup
```

Comment:
First off, I support adding this wrapper if it is something that benefits your work right now!

However, I'd really prefer to not document this so that we can eventually remove it again. That's because the "expensive" part with `DoubleCosets` and `DoubleCosetsNC` really is that creating GAP objects is much more expensive than creating Julia objects, and so is avoided. I mean, this is the implementation:
```gap
InstallGlobalFunction( DoubleCosets, function(G,U,V)
  if not IsSubset(G,U) and IsSubset(G,V) then
    Error("not contained");
  fi;
  return DoubleCosetsNC(G,U,V);
end );

InstallMethod(DoubleCosetsNC,"generic",true,
  [IsGroup,IsGroup,IsGroup],0,
function(G,U,V)
  return List(DoubleCosetRepsAndSizes(G,U,V),i->DoubleCoset(U,i[1],V,i[2]));
end);
```

But this is quite different with Julia objects. So what we should do ultimately (and I explicitly don't mean Simon in this PR, but rather Thomas and myself in a future PR) is to change `double_cosets` to call `DoubleCosetRepsAndSizes` instead of `DoubleCosets`/`DoubleCosetsNC`. GAP does not implement much functionality for double cosets anyway (basically: equality test, membership test, querying size and representative, `rand`, `rand_pseudo`, iteration/enumeration, printing, conversion to a list), and we can just re-implement all of this in a few dozen (or at most few hundred) lines of rather simple code. But initially, we can just store a representative and size; and an optional, "lazy" pointer to a GAP double coset object, which we only created when needed. And then we can re-implement functionality gradually.

We could also change `double_cosets` to return an iterator to make it even more "lazy", but that's separate step.

Reply:
Should I remove it from the exports?